core/src/main/scala/pureconfig/ConfigReader.scala

```diff
@@ -118,7 +118,7 @@ trait ConfigReader[A] {
 /**
  * Provides methods to create [[ConfigReader]] instances.
  */
-object ConfigReader extends BasicReaders with CollectionReaders with ExportedReaders {
+object ConfigReader extends BasicReaders with CollectionReaders with ProductReaders with ExportedReaders {
```

Comment:
I like the idea a lot, but I'd like to see unit tests before we approve it.

Reply:
The unit tests are generated the same way as the files themselves. While this can lead to a "who watches the watchmen" situation, the tests can actually be seen in the Travis log.

Reply:
Sorry I missed that.

Reply:
The tests are also cumbersome to write by hand without a generator. In 5161a2a, I ended up adding some extra cases to test not only the successful path but also that we're producing the correct failures in `ConfigReader`s built via the `forProductN` methods.

Reply:
@jcazevedo,

Thanks for switching to sbt-boilerplate. I find having separate `.template` files easier to understand.